libraries/AP_GPS/AP_GPS_VecNAV.cpp

```diff
@@ -0,0 +1,345 @@
+    count++;    
+    if(count == (DATA_LEN - HEADER - CRC_LEN)){      
+      //printf("Receive payload, count=%d\n",count);
+       //hal.console->printf("Receive payload, count=%d\n",count);
+      
+      msgState = 4;
+      count = 0;      
+    }
+
+    break;
+    
+  case 4:
+    //CRC
+    //printf("datasize = %d\n",packetsize);
+    dataPl[packetsize-1] = c;
+    packetsize++;
+    if(count == 0){
+      CRC = 0;
+      CRC = CRC | c;
+      count++;
+    }
+    else if(count == 1){
+      CRC = (CRC << 8) | c;
+      msgState = 5;
+      count = 0;
+      
+      //printf("Received checksum\n");
+    }
+
+    break;
+
+  case 5:
+    // Validate data    
+    CRCV = calculateCRC(dataPl,DATA_LEN);
+    msgState = 0;
+    count = 0;
+    if(CRCV == 0x0000){
+      //printf("Valid data obtained\n");
+      //hal.console->print("Valid data obtained");
+
+      p = dataPl + HEADER;
+      ExtractDataGroup1(p,msgG1);
+      
+      p = dataPl + HEADER + GROUP1_LEN;
+      ExtractDataGroup4(p,msgG4);
+
+      p = dataPl + HEADER + GROUP1_LEN + GROUP4_LEN;
+      ExtractDataGroup6(p,msgG6);
+      
+      return true;
+    }
+    else{      
+      return false;
+    }
+    
+    break;
+  }//end of switch
+
+  return false;
+}
+
+bool AP_GPS_VecNAV::read(void){
+
+    int16_t numc;
+    bool parsed = false;
+      
+    numc = port->available();
+
+    //hal.console->printf("Available data %d\n",numc);
+
+    while (numc--) {
+        
+        char c = port->read();
+        //hal.console->printf("%c",c);
+        parsed = ProcessGPSMessage(c,&msg1,&msg4,&msg6);
+
+        if(parsed){
+            
+            state.location.lng    = (int32_t) (msg6.PosLla[1]*1e7);
+            state.location.lat    = (int32_t) (msg6.PosLla[0]*1e7);
+            state.location.alt    = (int32_t) (msg6.PosLla[2] * 100); // altitude in centimeters
+
+            if(msg4.Fix == 0 || msg4.Fix == 1){
+                state.status = AP_GPS::NO_FIX;
+            }
+            else if(msg4.Fix == 2){
+                state.status = AP_GPS:: GPS_OK_FIX_2D;
+            }
+            else if(msg4.Fix == 3){
+                state.status = AP_GPS:: GPS_OK_FIX_3D;
+            }
+
+            //state.horizontal_accuracy = msg4.PosU[0]*100;
+            //state.vertical_accuracy = msg4.PosU[2]*100;
+            state.have_horizontal_accuracy = false;
+            state.have_vertical_accuracy = false;
+            
+            state.have_vertical_velocity = true;
+            state.velocity.x = msg6.VelNed[0];
+            state.velocity.y = msg6.VelNed[1];
+            state.velocity.z = msg6.VelNed[2];
+            state.ground_course = wrap_360(degrees(atan2f(state.velocity.y, state.velocity.x)));
+            state.ground_speed = norm(state.velocity.y, state.velocity.x);
+            state.have_speed_accuracy = true;
+            state.speed_accuracy = msg6.VelU;
+            state.num_sats = msg4.NumSats;
+            state.hdop     = 130;           
```

Comment:
If the DOP's aren't available (which they aren't) this should be 9999 rather then a fake/incorrect DOP.